libs/libcommon/tests/test_queue.py

```diff
@@ -383,3 +386,30 @@ def test_has_ttl_index_on_finished_at_field() -> None:
     ttl_index_name = ttl_index_names[0]
     assert ttl_index_name == "finished_at_1"
     assert Job._get_collection().index_information()[ttl_index_name]["expireAfterSeconds"] == QUEUE_TTL_SECONDS
+
+
+def increment(tmp_file: Path) -> None:
+    with open(tmp_file, "r") as f:
+        current = int(f.read() or 0)
+    with open(tmp_file, "w") as f:
+        f.write(str(current + 1))
+
+
+def locked_increment(tmp_file: Path) -> None:
+    with lock(key="test_lock", job_id=str(os.getpid())):
+        increment(tmp_file)
+
+
+def test_lock(tmp_path_factory: pytest.TempPathFactory, queue_mongo_resource: QueueMongoResource) -> None:
```

Comment:
good idea to have used the same database (Queue), we don't have to create another "resource".